### PR TITLE
bridge: Fix data injection

### PR DESCRIPTION
**Pull request by dperpeet**
Depending on the data and markers used, certain occurrences could be missed.

Follow-up for #4113

### CODE REVIEW

**Comment by stefwalter** [comment]
```
==32616== 56 (40 direct, 16 indirect) bytes in 1 blocks are definitely lost in loss record 573 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x409953: test_web_filter_multiple (test-webresponse.c:620)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_multiple
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 599 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x409AA9: test_web_filter_simple (test-webresponse.c:570)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_simple
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 600 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x409899: test_web_filter_multiple (test-webresponse.c:602)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_multiple
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 601 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x4098D7: test_web_filter_multiple (test-webresponse.c:608)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_multiple
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 602 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x409915: test_web_filter_multiple (test-webresponse.c:614)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_multiple
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 603 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x4089FE: test_web_filter_split (test-webresponse.c:662)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_split
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 604 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x408A3C: test_web_filter_split (test-webresponse.c:668)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_split
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 605 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x408A7A: test_web_filter_split (test-webresponse.c:674)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_split
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 72 (40 direct, 32 indirect) bytes in 1 blocks are definitely lost in loss record 606 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x408819: test_web_filter_shift (test-webresponse.c:731)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_shift
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
==32616== 104 (40 direct, 64 indirect) bytes in 1 blocks are definitely lost in loss record 758 of 804
==32616==    at 0x4C2AB80: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==32616==    by 0x5449610: g_malloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x545F22D: g_slice_alloc (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5418C44: g_array_sized_new (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x40CA45: cockpit_web_inject_new (cockpitwebinject.c:220)
==32616==    by 0x409749: test_web_filter_passthrough (test-webresponse.c:770)
==32616==    by 0x5467A90: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467C55: ??? (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FAA: g_test_run_suite (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x5467FE0: g_test_run (in /lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0)
==32616==    by 0x407431: main (test-webresponse.c:1246)
==32616== 
{
   <insert_a_suppression_name_here>
   Memcheck:Leak
   match-leak-kinds: definite
   fun:malloc
   fun:g_malloc
   fun:g_slice_alloc
   fun:g_array_sized_new
   fun:cockpit_web_inject_new
   fun:test_web_filter_passthrough
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   obj:/lib/x86_64-linux-gnu/libglib-2.0.so.0.4000.0
   fun:g_test_run_suite
   fun:g_test_run
   fun:main
}
PASS: test-config 2 /conf/test-strings
```

**Comment by dperpeet** [comment]
I forgot to unref the array.

File: src/common/cockpitwebinject.c [hunk 1]
```diff
@@ -84,70 +89,100 @@ cockpit_web_inject_push (CockpitWebFilter *filter,
...
-        {
-          function (func_data, block);
-        }
-      else
-        {
-          bytes = g_bytes_new_from_bytes (block, written, data_len - written);
-          function (func_data, bytes);
-          g_bytes_unref (bytes);
-        }
+      if (memcmp(mark, data + data_len - partial_len, partial_len) == 0)
```
Comment (stefwalter) [inline src/common/cockpitwebinject.c]: Missing space.

File: src/common/cockpitwebinject.c [hunk 2]
```diff
@@ -84,70 +89,100 @@ cockpit_web_inject_push (CockpitWebFilter *filter,
...
     }
 
-  if (written < data_len)
+    if (at >= data_len)
+      break;
+
+    /* if there are enough chars left, try to find a complete match */
+    if (self->injected < self->maximum &&
+        data_len >= (at + mark_len) &&
+        (pos = memmem(data + at, data_len - at, mark, mark_len)))
```
Comment (stefwalter) [inline src/common/cockpitwebinject.c]: Missing space.

File: src/common/cockpitwebinject.c [hunk 3]
```diff
@@ -84,70 +89,100 @@ cockpit_web_inject_push (CockpitWebFilter *filter,
...
+              /* our match can only grow longer */
+              g_array_index (self->partial_matches, gboolean, partial_len) = FALSE;
+              remaining_len = mark_len - partial_len;
+              /* our current block might be too short */
+              if (remaining_len > data_len)
+                {
+                  if (memcmp(mark + partial_len, data, data_len) == 0)
+                    g_array_index (self->partial_matches, gboolean, partial_len + data_len) = TRUE;
+                }
+              else if (memcmp(mark + partial_len, data, remaining_len) == 0)
```
Comment (stefwalter) [inline src/common/cockpitwebinject.c]: Needs an extra space

File: src/common/cockpitwebinject.c [hunk 4]
```diff
@@ -84,70 +89,100 @@ cockpit_web_inject_push (CockpitWebFilter *filter,
...
-              bytes = g_bytes_new_from_bytes (block, written, at - written);
-              function (func_data, bytes);
-              g_bytes_unref (bytes);
+              /* our match can only grow longer */
+              g_array_index (self->partial_matches, gboolean, partial_len) = FALSE;
+              remaining_len = mark_len - partial_len;
+              /* our current block might be too short */
+              if (remaining_len > data_len)
+                {
+                  if (memcmp(mark + partial_len, data, data_len) == 0)
```
Comment (stefwalter) [inline src/common/cockpitwebinject.c]: Needs a space.

**Comment by stefwalter** [comment]
One more test?

```
static void
test_web_filter_shift_three (TestCase *tc,
                             gconstpointer data)
{
  CockpitWebFilter *filter;
  const gchar *resp;
  GBytes *block;
  GBytes *inject;

  inject = bytes_static ("injected");
  filter = cockpit_web_inject_new ("foofn", inject, 1);
  cockpit_web_response_add_filter (tc->response, filter);
  g_object_unref (filter);
  g_bytes_unref (inject);

  cockpit_web_response_headers_full (tc->response, 200, "OK", -1, NULL);

  /* Total content is foofoofn and split after the first 4 characters */
  block = bytes_static ("foof");
  g_assert (cockpit_web_response_queue (tc->response, block) == TRUE);
  g_bytes_unref (block);
  block = bytes_static ("o");
  g_assert (cockpit_web_response_queue (tc->response, block) == TRUE);
  g_bytes_unref (block);
  block = bytes_static ("of");
  g_assert (cockpit_web_response_queue (tc->response, block) == TRUE);
  g_bytes_unref (block);
  block = bytes_static ("n");
  g_assert (cockpit_web_response_queue (tc->response, block) == TRUE);
  g_bytes_unref (block);
  cockpit_web_response_complete (tc->response);

  while (cockpit_web_response_get_state (tc->response) != COCKPIT_WEB_RESPONSE_COMPLETE)
    g_main_context_iteration (NULL, TRUE);

  resp = output_as_string (tc);
  g_assert_cmpint (cockpit_web_response_get_state (tc->response), ==, COCKPIT_WEB_RESPONSE_SENT);

  g_assert_cmpstr (resp, ==, "HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n"
                   "4\r\nfoof\r\n"
                   "1\r\no\r\n"
                   "2\r\nof\r\n"
                   "1\r\nn\r\n"
                   "8\r\ninjected\r\n"
                   "0\r\n\r\n");
}
```

File: src/common/cockpitwebinject.c (hunk 1 above)
Comment (dperpeet) [inline src/common/cockpitwebinject.c]: > Missing space.

added

File: src/common/cockpitwebinject.c (hunk 2 above)
Comment (dperpeet) [inline src/common/cockpitwebinject.c]: > Missing space.

added

File: src/common/cockpitwebinject.c (hunk 3 above)
Comment (dperpeet) [inline src/common/cockpitwebinject.c]: > Needs an extra space

added

File: src/common/cockpitwebinject.c (hunk 4 above)
Comment (dperpeet) [inline src/common/cockpitwebinject.c]: > Needs a space.

added

**Comment by dperpeet** [comment]
> One more test?

good idea - added & passed (only modified the comment slightly)

**Pull request closed without merging**